src/app/login/login.component.ts

```diff
@@ -12,19 +13,31 @@ export class LoginComponent implements OnInit {
   password: any = '';
   showFailedLoginInfo = false;
 
-  constructor(private router: Router, private authservice: AuthService) {}
+
+  constructor(private router: Router, private authservice: AuthService,private spinner: NgxSpinnerService) {}
 
   ngOnInit() {}
 
   login() {
-    if (this.username && this.password) {
-      console.log('Succesfully logged ' + this.username);
-      this.showFailedLoginInfo = false;
-      this.authservice.login(this.username, this.password);
-      this.router.navigate(['/courses']);
-    } else {
-      console.log('Unsuccesfull... ' + this.username);
-      this.showFailedLoginInfo = true;
-    }
+    this.spinner.show();
+    this.delay(3000).then(any => {
+
+        if (this.username && this.password) {
+          console.log('Succesfully logged ' + this.username);
+          this.showFailedLoginInfo = false;
+          this.authservice.login(this.username, this.password);
+          this.spinner.hide();
+          this.router.navigate(['/courses']);
+        } else {
+          console.log('Unsuccesfull... ' + this.username);
+          this.showFailedLoginInfo = true;
+          this.spinner.hide();
+        }
+    });
   }
+
+  async delay(ms: number) {
```

Comment:
no need for such a complicated method. You can use `timer` observable from rxjs

Reply:
changed